packages/arb-token-bridge-ui/src/pages/api/cctp/[type].ts

```diff
@@ -154,53 +194,70 @@ export default async function handler(
       l1ChainId === ChainId.Mainnet ? 'cctp-arb-one' : 'cctp-arb-goerli'
     )
 
-    const messagesSentQuery = gql(`{
-        messageSents(
-          where: {
-            sender: "${walletAddress}"
-            ${searchString ? `transactionHash_contains: "${searchString}"` : ''}
-          }
-          orderDirection: "desc"
-          orderBy: "blockTimestamp"
-          first: ${Number(pageSize)}
-          skip: ${Number(pageNumber) * Number(pageSize)}
-        ) {
-          attestationHash
-          blockNumber
-          blockTimestamp
-          id
-          message
-          nonce
-          sender
-          recipient
-          sourceDomain
-          transactionHash
-          amount
-        }
-      }`)
-
-    let messagesSentResult: ApolloQueryResult<{ messageSents: MessageSent[] }>
+    const messagesSentQueryFromWalletAddress = getMessageSents({
+      sender: walletAddress,
+      searchString,
+      pageSize,
+      pageNumber
+    })
+    const messagesSentQueryToWalletAddress = getMessageSents({
+      recipient: walletAddress,
+      searchString,
+      pageSize,
+      pageNumber
+    })
+    let messagesSentToWalletAddressResult: ApolloQueryResult<{
+      messageSents: MessageSent[]
+    }>
+    let messagesSentFromWalletAddressResult: ApolloQueryResult<{
+      messageSents: MessageSent[]
+    }>
     if (type === 'deposits') {
-      messagesSentResult = await l1Subgraph.query({ query: messagesSentQuery })
+      ;[
```

Comment:
Since queries are the same for both deposits and withdrawals, we could simplify it to:

```
const subgraph = type === 'deposits' ? l1Subgraph : l2Subgraph

;[messagesSentToWalletAddressResult, messagesSentFromWalletAddressResult] = await Promise.all([
  subgraph.query({
    query: messagesSentQueryToWalletAddress
  }),
  subgraph.query({
    query: messagesSentQueryFromWalletAddress
  })
])
```

Reply:
Done in #1093